crates/macro/src/derive_props/field.rs

```diff
@@ -90,12 +90,14 @@ impl PropField {
                 // Hacks to avoid misleading error message.
                 quote_spanned! {span=>
                     #name: {
-                        let none: Option<#ty> = None;
                         match true {
-                            false => none,
-                            true => Some(#default())
+                            #[allow(unreachable_code)]
+                            false => {
+                                let __unreachable: #ty = ::std::unreachable!();
+                                __unreachable
```

Comment:
Do you need an assign here? I think you can just do `false => ::std::unreachable!()` right?

Reply:
The `false` arm must return the correct type, otherwise Rust would think that it has the same type as `#default()`, and might show that misleading message: "help: try using a conversion method: `"foo".to_string()`".

Reply:
Hmm your explanation makes sense but I had it working with unimplemented like that. This is fine though!

Reply:
Sorry, I was wrong about that. I did not test it before leaving the comment. It seems that `false => ::std::unreachable!()` does work.